llvm/lib/Target/PowerPC/AsmParser/PPCAsmParser.cpp

```diff
@@ -1252,14 +1259,30 @@ void PPCAsmParser::processInstruction(MCInst &Inst,
 static std::string PPCMnemonicSpellCheck(StringRef S, const FeatureBitset &FBS,
                                          unsigned VariantID = 0);
 
+// Check that the register+immediate memory operand is in the right position and
+// is expected by the instruction
+static bool validateMemOp(const OperandVector &Operands, bool isMemriOp) {
+  for (size_t idx = 0; idx < Operands.size(); ++idx) {
+    const PPCOperand &Op = static_cast<const PPCOperand &>(*Operands[idx]);
+    if ((idx == 3 && Op.isMemOpBase() != isMemriOp) ||
+        (idx != 3 && Op.isMemOpBase())) {
+        return false;
+    }
```

Comment:
Logic can be expressed more succinctly:
```suggestion
    if (Op.isMemOpBase() != (idx == 3 && isMemriOp))
      return false;
```